WireDomain/Sources/WireDomain/Repositories/Connections/ConnectionsLocalStore.swift

```diff
@@ -0,0 +1,389 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreData
+import Foundation
+import WireAPI
+import WireDataModel
+
+// sourcery: AutoMockable
+protocol ConnectionsLocalStoreProtocol {
+
+    func storeConnection(
+        _ connectionPayload: Connection
+    ) async throws
+
+    func deleteFederationConnection(
+        with domain: String
+    ) async throws
+
+    func removeFederationConnection(
+        between domain: String,
+        and otherDomain: String
+    ) async
```

Comment:
These connections are not quite the same. `Connection` is the connection between two users (via a connection request). The connection between federating domains is the ability to communicate between those domains.

What do you think about moving the federation logic up into the event processor?

Reply:
It makes sense, fixed in https://github.com/wireapp/wire-ios/pull/1918/commits/c77f7a4babb947cd5f43ea7e091187dbddd08c68